### ISSUE TITLE
Cypress 12 Support

### ISSUE THREAD

**Issue by BFMeenink** [badeball/cypress-cucumber-preprocessor#899]
### Current behavior

As of December 6th, 2022, Cypress 12 has been released. 
(https://www.cypress.io/blog/2022/12/06/announcing-cypress-12/)

When upgrading to Cypress 12, npm install gives:
```
cypress-e2e_1               | npm ERR! code ERESOLVE
cypress-e2e_1               | npm ERR! ERESOLVE unable to resolve dependency tree
cypress-e2e_1               | npm ERR! 
cypress-e2e_1               | npm ERR! While resolving: admin-portal-e2e@0.0.0
cypress-e2e_1               | npm ERR! Found: cypress@12.0.1
cypress-e2e_1               | npm ERR! node_modules/cypress
cypress-e2e_1               | npm ERR!   dev cypress@"12.0.1" from the root project
cypress-e2e_1               | npm ERR! 
cypress-e2e_1               | npm ERR! Could not resolve dependency:
cypress-e2e_1               | npm ERR! peer cypress@"^8.0.0 || ^9.0.0 || ^10.0.0 || ^11.0.0" from @badeball/cypress-cucumber-preprocessor@14.0.0
```

### Desired behavior

To be compatible with Cypress 12.

### Test code to reproduce

Can provide partial of package.json if so desired

### Versions

* **Cypress version**: 12
* **Preprocessor version**: 14.0.0
* **Node version**: v16.18.1

### Checklist

- [X] I've read the [FAQ](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/docs/faq.md).
- [X] I've read [Instructions for logging issues](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/CONTRIBUTING.md#instructions-for-logging-issues).
- [X] I'm not using `cypress-cucumber-preprocessor@4.3.1` (package name has changed and it is no longer the most recent version, see [#689](https://github.com/badeball/cypress-cucumber-preprocessor/issues/689)).

**Comment by remarkablemark** [comment]
Please take a look at #902

**Comment by badeball** [comment]
Fixed with v15.0.0.
